core/src/main/java/org/elasticsearch/search/aggregations/bucket/terms/LongTerms.java

```diff
@@ -126,26 +99,36 @@ public XContentBuilder toXContent(XContentBuilder builder, Params params) throws
             builder.endObject();
             return builder;
         }
+
+        public DoubleTerms.Bucket convertToDouble() {
```

Comment:
Could you explain why this method had to be moved to here? I am a little uneasy with the LongTerms implementation having to know about the DoubleTerms implementation and containing logic to convert between the two. This seems like something the DoubleTermsAggregator should be doing?